### PR TITLE
Add test cases for log mediator

### PR DESCRIPTION
**Pull request by Dilshani**
Tests will cover followings in log mediator
- Levels
- Categories

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("LogMediatorLevelAndCategoryTestProxy"),
+                        null, "WSO2");
+        Assert.assertTrue(response.toString().contains("WSO2"));
+        logs = logViewer.getAllRemoteSystemLogs();
+    }
+
+    @Test(groups = "wso2.esb", description = "Test debug level log")
+    public void testDebugLevelLogs() throws Exception {
+        Boolean isDebugLogAvailable = false;
+        //LogEvent[] getLogsDebug = logViewer.getLogs("DEBUG", "LogMediator", "", "");
```
Comment (Manuri) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Can we remove this if this line is not needed?

Comment (Dilshani) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Sure, Will do

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                isTraceLogAvailable = true;
+            }
+        }
+        Assert.assertTrue(isTraceLogAvailable, "TRACE Log not found. Trace logs not working properly");
+    }
+
+    @Test(groups = "wso2.esb", description = "Test info level log")
+    public void testInfoLevelLogs() throws Exception {
+        Boolean isInfoLogAvailable = false;
+        for (LogEvent logEvent : logs) {
```
Comment (prabushi) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Can't we move this 'for' loop to a common method? It seems same functionality is getting repeated.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        logAdmin = new LoggingAdminClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewer = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+        logViewer.clearLogs();
+        verifyProxyServiceExistence("LogMediatorLevelAndCategoryTestProxy");
+        logAdmin.updateLoggerData("org.apache.synapse.mediators.builtin",
+                LoggingAdminClient.LogLevel.TRACE.name(),true, false);
+        response = axis2Client
+                .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("LogMediatorLevelAndCategoryTestProxy"),
+                        null, "WSO2");
+        Assert.assertTrue(response.toString().contains("WSO2"));
```
Comment (manoramahp) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Add assert message here

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        for (LogEvent logEvent : logs) {
+            String message = logEvent.getMessage();
+            if (message.contains("*****LOGGING IN DEBUG CATEGORY - CUSTOM LEVEL*****")) {
+                isDebugLogAvailable = true;
+            }
+        }
+        Assert.assertTrue(isDebugLogAvailable, "DEBUG Log not found. Debug logs not working properly");
+    }
+
+    //This test case disabled due to issue reported as "[EI 620][Integration Profile] Trace logs not going to indicate
```
Comment (ramith) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: should this be java doc comment?

Comment (Dilshani) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Yeah, it should be. Will update

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java (hunk 3 above)
Comment (Dilshani) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Noted

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java (hunk 2 above)
Comment (Dilshani) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/log/LogMediatorLevelsAndCategoryTestCase.java]: Yes, Will do.

**Pull request merged**